### PR TITLE
Fixed Asset Browser operation

### PR DESCRIPTION
**Pull request by jjjoness**
Signed-off-by: John Jones-Steele <82226755+jjjoness@users.noreply.github.com>

## What does this PR do?

Fixes bug #11592 

## How was this PR tested?

Testing in local branch

### CODE REVIEW

**Comment by gadams3** [comment]
@jjjoness Thank you for taking a look at this. Can you please give more context on how these changes alter double click renaming? I think all of the item views have built in settings and signals to handle the commonly expected behavior for editing and renaming items.

I have not tried this in the main editor asset browser. But in the material editor the repro steps for what I was experiencing are as follows:
- open the material editor
- open the material editor asset browser
- double click on a material or material type source file
- expected behavior: material editor opens new document for that file
- actual behavior: asset browser initiates renaming operation

thanks again!

**Comment by jjjoness** [comment]
@gadams3 Sorry for the delay in getting back, it has been a holiday in the UK. I am checking my solution against the main branch to see what I'm missing.

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp [hunk 1]
```diff
@@ -491,6 +491,14 @@ namespace AzToolsFramework
             }
         }
 
+        void AssetBrowserTreeView::mouseDoubleClickEvent(QMouseEvent* ev)
+        {
+            if (ev->button() == Qt::MouseButton::LeftButton)
+            {
+                emit doubleClickedSignal(QModelIndex());
```
Comment (alexmontAmazon) [inline Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp:498]: why have the model index as a signal parameter if you're always emitting a default constructed index? Do you have plans to use this signal more robustly in the future?

File: Code/Editor/AzAssetBrowser/AzAssetBrowserWindow.cpp [hunk 2]
```diff
@@ -152,7 +152,9 @@ AzAssetBrowserWindow::AzAssetBrowserWindow(QWidget* parent)
     connect(m_ui->m_assetBrowserTreeViewWidget, &AzAssetBrowser::AssetBrowserTreeView::selectionChangedSignal,
         this, &AzAssetBrowserWindow::SelectionChangedSlot);
 
-    connect(m_ui->m_assetBrowserTreeViewWidget, &QAbstractItemView::doubleClicked, this, &AzAssetBrowserWindow::DoubleClickedItem);
```
Comment (alexmontAmazon) [inline Code/Editor/AzAssetBrowser/AzAssetBrowserWindow.cpp:155]: was this doubleClicked not firing at all, or was it in addition to the activated signal? Could you comment this please?

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp [hunk 3]
```diff
@@ -491,6 +491,14 @@ namespace AzToolsFramework
...
+            }
+        }
```
Comment (alexmontAmazon) [inline Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp:500]: you're neither calling accept() nor ignore on the event here. Do you want it to propagate further?
https://doc.qt.io/qt-6/qmouseevent.html

**Review by alexmontAmazon — COMMENTED** [review]
needs a bit more clarity / commenting, but I'm not against it in spirit. I think it might be doable with the existing signals and accepting/ignoring earlier in the chain, though.

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp (hunk 1 above)
Comment (jjjoness) [reply Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp]: Totally rewritten.

File: Code/Editor/AzAssetBrowser/AzAssetBrowserWindow.cpp (hunk 2 above)
Comment (jjjoness) [reply Code/Editor/AzAssetBrowser/AzAssetBrowserWindow.cpp]: Totally rewritten.

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp (hunk 3 above)
Comment (jjjoness) [reply Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp]: Totally rewritten.

**Review by alexmontAmazon — APPROVED** [review]
Nice!

**Review by gadams3 — APPROVED** [review]

**Pull request merged**